### PR TITLE
Creation of addon

### PR DESCRIPTION
**Pull request by DarkRails**
This addon add the /is fly command to let player fly on their own island, when they leave their island they receive a warn message to say fly mode will be disabled soon, if player is not back on his island after a configurable amount of seconds, his fly is disabled. 

PS: This is my first PR sorry if I have done something wrong.

### CODE REVIEW

**Comment by Poslovitch** [comment]
Hi. I'm going to review this PR, but thanks anyway :)

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>fr.zeltiamc</groupId>
```
Comment (Poslovitch) [inline pom.xml]: IslandFly is an official addon, please use the correct groupId.
```suggestion
    <groupId>world.bentobox</groupId>
```

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    <artifactId>islandfly</artifactId>
```
Comment (Poslovitch) [inline pom.xml]: artifactId should be identical to the name
```suggestion
    <artifactId>IslandFly</artifactId>
```

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        <repository>
+            <id>codemc-repo</id>
+            <url>https://repo.codemc.org/repository/maven-public/</url>
+        </repository>
+    </repositories>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.destroystokyo.paper</groupId>
+            <artifactId>paper-api</artifactId>
```
Comment (Poslovitch) [inline pom.xml]: An official addon must be using spigot-api, not paper's. Certainly PaperSpigot does bring some improvements, but these come with incompatibilities. 
```suggestion
            <artifactId>spigot-api</artifactId>
```

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            <version>1.13.2-R0.1-SNAPSHOT</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>world.bentobox</groupId>
+            <artifactId>bentobox</artifactId>
+            <version>0.15.0-SNAPSHOT</version>
```
Comment (Poslovitch) [inline pom.xml]: I'd suggest you to rather use 0.14.1. It is the latest "stable" version.
 ```suggestion
            <version>0.14.1</version>
```

File: src/main/java/bentobox/addon/islandfly/listeners/FlyLogoutListener.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import org.bukkit.event.player.PlayerQuitEvent;
+
+public class FlyLogoutListener implements Listener {
+
+    @EventHandler
+    public void onLogout(final PlayerQuitEvent event){
+
+        final Player player = event.getPlayer();
+
+        if(!event.getPlayer().getAllowFlight()) return;
```
Comment (Poslovitch) [inline src/main/java/bentobox/addon/islandfly/listeners/FlyLogoutListener.java]: You set up a `Player` variable right just above, why not using it there?

File: src/main/java/bentobox/addon/islandfly/listeners/FlyLogoutListener.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        if(event.getPlayer().hasPermission("islandfly.bypass")) return;
```
Comment (Poslovitch) [inline src/main/java/bentobox/addon/islandfly/listeners/FlyLogoutListener.java]: I'd rather do these checks in the same line:
```suggestion
        if(!player.getAllowFlight() || player.hasPermission("islandfly.bypass")) return;
```

File: src/main/resources/addon.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+name: IslandFly
+version: ${project.version}
+main: bentobox.addon.islandfly.IslandFlyAddon
+authors: [Poslovitch, DarkRails]
```
Comment (Poslovitch) [inline src/main/resources/addon.yml]: No need to put me as an author there.

File: src/main/resources/addon.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+name: IslandFly
+version: ${project.version}
+main: bentobox.addon.islandfly.IslandFlyAddon
```
Comment (Poslovitch) [inline src/main/resources/addon.yml]: Ah! You're not using the "package format" for official addons.
It should rather be:
```suggestion
main: world.bentobox.islandfly.IslandFlyAddon
```

File: src/main/java/bentobox/addon/islandfly/listeners/FlyListener.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+        // Player already flying
+        final User user = User.getInstance(playerUUID);
+        if(!user.getPlayer().getAllowFlight()) return;
+        // Bypass permission
+        if(user.hasPermission("islandfly.bypass")) return;
+
+        // Alert player fly will be disabled
+        final int flyTimeout = settings.getFlyTimeout();
+        user.sendMessage("islandfly.fly-outside-alert", "[timeout]", String.valueOf(flyTimeout));
```
Comment (Poslovitch) [inline src/main/java/bentobox/addon/islandfly/listeners/FlyListener.java]: You should use the "text variable" `[number]`, like this:
```suggestion
        user.sendMessage("islandfly.fly-outside-alert", TextVariables.NUMBER, String.valueOf(flyTimeout));
```

It's not that important though, it's mainly about "consistency".
I let you choose for this one.

File: src/main/resources/locales/fr-FR.yml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+islandfly:
+  fly-outside-alert: "Vous êtes hors de votre île, vous ne pourrez plus voler dans [timeout] secondes."
+  disable-fly: "Votre mode volant vient d'être désactivé."
+  enable-fly: "Votre mode volant vient d'être activé."
```
Comment (Poslovitch) [inline src/main/resources/locales/fr-FR.yml]: A better translation for the two above lines would be:
```
disable-fly: "Vous ne pouvez plus voler."
enable-fly: "Vous pouvez désormais voler."
```
Mais là encore, je te laisse le choix 😛.

File: src/main/resources/locales/en-US.yml [hunk 11]
```diff
@@ -0,0 +1,6 @@
+islandfly:
+  fly-outside-alert: "You are outside your island fly mode will be disabled in [timeout] seconds."
+  disable-fly: "Your fly mode has been disabled."
+  enable-fly: "Your fly mode has been enabled."
+  command:
+    description: "Allow you to fly on your island."
```
Comment (Poslovitch) [inline src/main/resources/locales/en-US.yml]: Command description shouldn't start with an uppercase:
```suggestion
    description: "allows you to fly on your island"
```

File: src/main/resources/locales/fr-FR.yml [hunk 12]
```diff
@@ -0,0 +1,6 @@
...
+  command:
+    description: "Vous permet de voler sur votre île."
```
Comment (Poslovitch) [inline src/main/resources/locales/fr-FR.yml]: Same:
```suggestion
    description: "vous permet de voler sur votre île"
```

File: src/main/java/bentobox/addon/islandfly/IslandFlyAddon.java [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+        // Load configuration
+        this.settings = new FlySettings(this);
+        // Register Listeners
+        final PluginManager pluginManager = Bukkit.getPluginManager();
+        pluginManager.registerEvents(new FlyListener(this), this.getPlugin());
+        // Register listener to disable fly on logout if activated
+        if(settings.isFlyDisabledOnLogout())
+            pluginManager.registerEvents(new FlyLogoutListener(), this.getPlugin());
+        // BSkyBlock hook in
+        this.getPlugin().getAddonsManager().getAddonByName("BSkyBlock").ifPresent(acidIsland -> {
```
Comment (tastybento) [inline src/main/java/bentobox/addon/islandfly/IslandFlyAddon.java]:
```suggestion
        this.getPlugin().getAddonsManager().getAddonByName("BSkyBlock").ifPresent(bskyblock -> {
```

File: src/main/resources/addon.yml (hunk 8 above)
Comment (DarkRails) [reply src/main/resources/addon.yml]: I changed the package name to world.bentobox

File: src/main/resources/locales/fr-FR.yml (hunk 10 above)
Comment (DarkRails) [reply src/main/resources/locales/fr-FR.yml]: I changed these two lines,.

Effectivement sa me parait mieux :D

File: src/main/resources/locales/en-US.yml [hunk 14]
```diff
@@ -0,0 +1,6 @@
+islandfly:
+  fly-outside-alert: "You are outside your island fly mode will be disabled in [timeout] seconds."
```
Comment (tastybento) [inline src/main/resources/locales/en-US.yml]:
```suggestion
  fly-outside-alert: "You are outside your island so fly mode will be disabled in [timeout] seconds."
```

File: src/main/resources/addon.yml [hunk 15]
```diff
@@ -0,0 +1,6 @@
+name: IslandFly
+version: ${project.version}
+main: world.bentobox.islandfly.IslandFlyAddon
+authors: [Poslovitch, DarkRails]
```
Comment (tastybento) [inline src/main/resources/addon.yml]:
```suggestion
authors: [DarkRails]
```
💯

**Comment by Poslovitch** [comment]
We'd now need to test the addon a bit before approving this PR.

**Comment by Poslovitch** [comment]
Oops! Forgot to merge the PR.

**Pull request merged**